sql/src/test/java/org/apache/apex/malhar/sql/SerDeTest.java

```diff
@@ -0,0 +1,223 @@
+import java.io.IOException;
+import java.util.Properties;
+
+import org.junit.Test;
+
+import org.apache.apex.malhar.kafka.KafkaSinglePortInputOperator;
+import org.apache.apex.malhar.kafka.KafkaSinglePortOutputOperator;
+import org.apache.apex.malhar.sql.table.CSVMessageFormat;
+import org.apache.apex.malhar.sql.table.Endpoint;
+import org.apache.apex.malhar.sql.table.FileEndpoint;
+import org.apache.apex.malhar.sql.table.KafkaEndpoint;
+import org.apache.apex.malhar.sql.table.StreamEndpoint;
+
+import org.apache.commons.io.FileUtils;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+
+import com.datatorrent.contrib.formatter.CsvFormatter;
+import com.datatorrent.contrib.parser.CsvParser;
+import com.datatorrent.stram.plan.logical.LogicalPlan;
+
+public class SerDeTest
+{
+  @Test
+  public void testSQLWithApexFactory() throws IOException, ClassNotFoundException
+  {
+    File modelFile = new File("src/test/resources/model/model_file_csv.json");
+    String model = FileUtils.readFileToString(modelFile);
+
+    LogicalPlan dag = new LogicalPlan();
+    SQLExecEnvironment.getEnvironment()
+        .withModel(model)
+        .executeSQL(dag, "SELECT STREAM ROWTIME, PRODUCT FROM ORDERS");
+
+    dag.validate();
+  }
+
+  @Test
+  public void testSQLWithAPI() throws ClassNotFoundException, IOException
+  {
+    LogicalPlan dag = new LogicalPlan();
+
+    String schema = "{\"separator\":\",\",\"quoteChar\":\"\\\"\",\"fields\":[" +
+        "{\"name\":\"RowTime\",\"type\":\"Date\",\"constraints\":{\"format\":\"dd/MM/yyyy hh:mm:ss Z\"}}," +
+        "{\"name\":\"id\",\"type\":\"Integer\"},{\"name\":\"Product\",\"type\":\"String\"}," +
+        "{\"name\":\"units\",\"type\":\"Integer\"}]}";
+
+    Endpoint endpoint = new FileEndpoint("/tmp/test", new CSVMessageFormat(schema));
+
+    SQLExecEnvironment.getEnvironment()
+        .registerTable("ORDERS", endpoint)
+        .executeSQL(dag, "SELECT STREAM FLOOR(ROWTIME TO HOUR), SUBSTRING(PRODUCT, 0, 5) FROM ORDERS WHERE id > 3");
+
+    dag.validate();
+  }
+
+  @Test
+  public void testSQLSelectInsertWithAPI() throws IOException, ClassNotFoundException
+  {
+    LogicalPlan dag = new LogicalPlan();
+
+    String schemaIn = "{\"separator\":\",\",\"quoteChar\":\"\\\"\",\"fields\":[" +
+        "{\"name\":\"RowTime\",\"type\":\"Date\",\"constraints\":{\"format\":\"dd/MM/yyyy hh:mm:ss\"}}," +
+        "{\"name\":\"id\",\"type\":\"Integer\"}," +
+        "{\"name\":\"Product\",\"type\":\"String\"}," +
+        "{\"name\":\"units\",\"type\":\"Integer\"}]}";
+    String schemaOut = "{\"separator\":\",\",\"quoteChar\":\"\\\"\",\"fields\":[" +
+        "{\"name\":\"RowTime\",\"type\":\"Date\",\"constraints\":{\"format\":\"dd/MM/yyyy hh:mm:ss\"}}," +
+        "{\"name\":\"Product\",\"type\":\"String\"}]}";
+
+    SQLExecEnvironment.getEnvironment()
+        .registerTable("ORDERS", new FileEndpoint("/tmp/test", new CSVMessageFormat(schemaIn)))
```

Comment:
Not a real test. The app does not run here. The test only does dag validation and serde check. Again for clarity, I've changed this to dummyFilePath.